Backend/dev/test-tool/local-api/server.py

```
@@ -1065,11 +1066,85 @@ def _pick_best(local_ip: str, aws_ip: str) -> str:
 
 
+# ── Devbox auto-assignment ────────────────────────────────────────────────────
+
+DEVBOX_ID_FILE = PROJECT_ROOT / ".devbox-id.json"
+
+def _parse_ram_pct(usage: dict) -> float:
+    """'27.5Gi (88%)' -> 88.0. Unknown/missing -> 100.0 (least preferred)."""
+    try:
+        m = re.search(r"\((\d+(?:\.\d+)?)%\)", str((usage or {}).get("ram", "")))
+        return float(m.group(1)) if m else 100.0
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)